xinference/model/llm/utils.py

```diff
@@ -606,7 +606,34 @@ def _tool_calls_completion(cls, model_family, model_uid, c, tools):
                 f"Model {model_family.model_name} is not support tool calls."
             )
         logger.debug("Tool call content: %s, func: %s, args: %s", content, func, args)
+        return content, func, args
 
+    @classmethod
+    def _tools_token_filter(cls, model_family):
```

Comment:
I'm confused that what's this function for? It seems that if `family in ["qwen-chat", "qwen1.5-chat"]`, judge that whether `tokens.endswith("\nFinal Answer:")`. Why need this complicated function?

Reply:
I'd like to filter out Qwen's react thought process, i.e. "Thought, Action, Observe" and just keep what comes after "Final Answer:".

Reply:
Please make the code more cleaner. It is so complicated here. The code here is much less readable.

Reply:
If you don't think this is necessary, we can simply remove this function, as well as the relevant lines in vllm/core.py